lib/forgetsy.rb

```diff
@@ -1,44 +1,53 @@
-require 'active_support/core_ext'
-
-require 'forgetsy/set'
-require 'forgetsy/delta'
-
-require 'redis/namespace'
+# frozen_string_literal: true
+require "forgetsy/set"
+require "forgetsy/delta"
+require "logger"
+require "redis"
 
 module Forgetsy
-  extend self
-
   # Accepts:
-  #   1. A 'hostname:port' string
-  #   2. A 'hostname:port:db' string (to select the Redis db)
-  #   3. A 'hostname:port/namespace' string (to set the Redis namespace)
-  #   4. A redis URL string 'redis://host:port'
-  #   5. An instance of `Redis`, `Redis::Client`, `Redis::DistRedis`,
-  #      or `Redis::Namespace`.
-  def redis=(server)
-    if server.respond_to? :split
-      if server["redis://"]
-        redis = Redis.connect(:url => server, :thread_safe => true)
+  #   1. An instance of `Redis`, `Redis::Client`, `Redis::DistRedis`,
+  #      `Redis::Namespace`, etc. (i.e. an existing Redis client instance)
+  #   2. A hash of options to pass through to `Redis.new`. If a `:namespace`
+  #      key is also provided, that will not be passed through to `Redis.new`
+  #      but rather used to initialize a `Redis::Namespace`.
+  def self.redis=(options)
+    if options.is_a?(Hash)
+      namespace = options.delete(:namespace)
+      client = Redis.new(options)
+      if namespace
+        begin
+          require "redis/namespace"
```

Comment:
This will need to be added as a runtime dependency as this will just break in production currently.